staging/src/k8s.io/apiserver/pkg/storage/value/transformer.go

```diff
@@ -189,7 +192,11 @@ func (t *prefixTransformers) TransformFromStorage(data []byte, context Context)
 	if err := errors.Reduce(errors.NewAggregate(errs)); err != nil {
 		return nil, false, err
 	}
-	RecordTransformation("from_storage", "unknown", start, t.err)
+	RecordTransformation(
+		"from_storage",
+		"unknown",
+		start,
+		status.Error(codes.Internal, t.err.Error()))
```

Comment:
I think it may make more sense to replace the error in the constructor:

```go
func NewPrefixTransformers(err error, transformers ...PrefixTransformer) Transformer {
	if err == nil {
		err = fmt.Errorf("the provided value does not match any of the supported transformers")
	}
	return &prefixTransformers{
		transformers: transformers,
		err:          status.Error(codes.Internal, err.Error()),
	}
}
```

Reply:
Done.